tests/test_wordpress_api.py

```diff
@@ -27,245 +20,182 @@ def raise_for_status(*args):
 
 
 class TestWordPressApi(unittest.TestCase):
-    @patch("canonicalwebteam.http.CachedSession.get")
-    def test_getting(self, get):
-
-        get.return_value = MockResponse()
-
-        article = api.get_article(slug="test")
-        get.assert_called_once_with(
-            "https://admin.insights.ubuntu.com/wp-json/wp/v2/"
-            + "posts?slug=test&tags=&tags_exclude="
-        )
-        self.assertEqual(article, "hello_test")
-
-    @patch("canonicalwebteam.http.CachedSession.get")
-    def test_excluding_article(self, get):
-
-        get.return_value = MockResponse()
-
-        article = api.get_article(slug="test", tags_exclude=[1234, 5678])
-        get.assert_called_once_with(
-            "https://admin.insights.ubuntu.com/"
-            + "wp-json/wp/v2/posts?slug=test&tags=&tags_exclude=1234,5678"
-        )
-        self.assertEqual(article, "hello_test")
-
-    @patch("canonicalwebteam.http.CachedSession.get")
-    def test_including_article(self, get):
-
-        get.return_value = MockResponse()
-
-        article = api.get_article(slug="test", tags=[1234, 5678])
-        get.assert_called_once_with(
-            "https://admin.insights.ubuntu.com/"
-            + "wp-json/wp/v2/posts?slug=test&tags=1234,5678"
-            + "&tags_exclude="
-        )
-        self.assertEqual(article, "hello_test")
-
-    @patch("canonicalwebteam.http.CachedSession.get")
-    def test_including_and_excluding_article(self, get):
-
-        get.return_value = MockResponse()
-
-        article = api.get_article(
-            slug="test", tags=[1234, 5678], tags_exclude=[9876]
+    @vcr.use_cassette("fixtures/vcr_cassettes/articles_with_metadata.yaml")
+    def test_get_articles_with_metadata(self):
+        # This function is similar to get_articles, plus the metadata
+        # therefore all other functionality is tested via test_get_articles
```

Comment:
no it's not similar.

You are getting the result of the function which is a correct test. This comment can be removed.

Reply:
Could you explain this? I could still provide a myriad of parameters here, which @nottrobin already noticed, and needs a refactoring to stop using the `**kwargs` pass through to `build_get_articles_url`.

The comment is supposed to let the developer know that all the parameter testing is going to happen in the next test, and thus we need not replicate all of it here.

The only addition here is the metadata. I think this also shows that this part of the code could potentially needs some refactoring. F.e. add yet another parameter that enables metadata?